src/Writers/Writers.jl

```diff
@@ -18,23 +18,26 @@ import ..Documenter:
 
 using Compat
 
-# Driver method for document rendering.
-# -------------------------------------
+"""
+A parametric type that allows us to use multiple dispatch to pick the appropriate
+writer for each output format.
 
+The parameter `f` should be an instance of the [`Formats.Format`](@ref) enumeration.
 """
-Writes a [`Documents.Document`](@ref) object to `build` directory in specified file format.
+immutable Writer{f} end
+Writer(f::Formats.Format) = Writer{f}()
+
 """
-function render(doc::Documents.Document)
-    mime = Formats.mimetype(doc.user.format)
-    for (src, page) in doc.internal.pages
-        open(Formats.extension(doc.user.format, page.build), "w") do io
-            for elem in page.elements
-                node = page.mapping[elem]
-                render(io, mime, node, page, doc)
-            end
-        end
-    end
-end
+Writes a [`Documents.Document`](@ref) object to `.user.build` directory in
+the format specified in `.user.format`.
+
+The method should be overloaded in each writer as
+
+    render(::Writer{format}, doc)
+
+where `format` is one of the values of the [`Formats.Format`](@ref) enumeration.
+"""
+render(doc::Documents.Document) = render(Writer(doc.user.format), doc)
```

Comment:
I intend to use user-defined formats to prototype the HTML code "out of tree", so to speak. And in any case, if we're going to have an `if`, might as well let the type system handle it. A `Selector` makes 3 lines of code 14, so I feel that the current one is preferred.